sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -275,8 +276,37 @@ object LogicalPlanIntegrity {
    * Some plan transformers (e.g., `RemoveNoopOperators`) rewrite logical
    * plans based on this assumption.
    */
-  def checkIfExprIdsAreGloballyUnique(plan: LogicalPlan): Boolean = {
-    checkIfSameExprIdNotReused(plan) && hasUniqueExprIdsForOutput(plan)
+  def validateExprIdUniqueness(plan: LogicalPlan): Unit = {
```

Comment:
what do you think about giving each of these validations a name? then we could report the name in the error message. One way to easily implement it could be to just push the name on some global stack that we can reference in the event of an exception, then we don't have to do any weird exception wranging.

Reply:
We can add sub error classes in the validation failure error class. But I'd like to do it later as naming is always a hard problem and may take a lot of time to reach a consensus.

Reply:
OK, that's fine. Maybe leave a TODO then.